src/main/scala/strawman/collection/Iterator.scala

```diff
@@ -118,6 +120,44 @@ trait Iterator[+A] extends IterableOnce[A] { self =>
       else Iterator.empty.next()
   }
 
+  /*
+   * Implemented by means of a buffer to keep track of the last n elements during iteration.
+   */
+  def takeRight(n: Int): Iterator[A] = {
+    if (n == 0) Iterator.empty
+    else {
+      // Return an iterator that iterates over the elements via a buffer
+      new Iterator[A]() {
+        private[this] var index = 0
+        private[this] var count = 0
+        // Use a lazy val for the buffer to make sure initialization is done only if needed and at most once
+        private[this] lazy val buffer = {
+          // Iterate over all elements while keeping track of the last n
+          var buf = ArrayBuffer[A]()
+          while (self.hasNext) {
+            if (index >= buf.length) buf += self.next()
+            else buf(index) = self.next()
+            index = if ((index + 1) >= n) 0 else index + 1
+            if (count < n) count += 1
+          }
+          // Adjust the starting index if needed
+          index = if (index >= buf.length) 0 else index
+          buf
+        }
+        override def hasNext: Boolean = {
```

Comment:
This is just a minor comment on style: I prefer not to have `override` explicitly written unless I really override the member. In this case, we just implement an abstract member, we don’t override it.